lib/puppet/type/component.rb

```diff
@@ -71,6 +71,10 @@ def to_s
     reference.to_s
   end
 
+  # Overrides the default implementation to do nothing
```

Comment:
Would be nice to have a bit more description about what's special about component. Rough draft:

```
# This type is used to represent a Class or Define before the catalog is converted from the containment graph to the relationship graph. As such, it contains data (its class/define parameters), but does not have actual parameters or properties at the Type level. We can simply ignore anything flagged as sensitive here, since the contained resources will handle that sensitivity themselves. There is no risk of this information leaking into reports, since no Component instances survive the graph transmutation.
```
```